### PR TITLE
Support for Image Builder - AWS sub-accounts

### PR DESCRIPTION
**Pull request by kkeshavamurthy**
PR for Issue #19630 

I was not sure if there were any naming convention used for boskos resource types, so picked the most descriptive one.

CC: @codenrhoden @detiber @justinsb

### CODE REVIEW

File: config/prow/cluster/boskos-janitor.yaml [hunk 1]
```diff
@@ -51,7 +51,7 @@ spec:
         image: gcr.io/k8s-staging-boskos/janitor:v20201002-ae3497d
         args:
         - --boskos-url=http://boskos.test-pods.svc.cluster.local.
-        - --resource-type=gce-project,gpu-project,ingress-project,istio-project,scalability-presubmit-project,scalability-project,node-e2e-project
+        - --resource-type=gce-project,gpu-project,ingress-project,istio-project,scalability-presubmit-project,scalability-project,node-e2e-project,image-builder-aws-project
```
Comment (detiber) [inline config/prow/cluster/boskos-janitor.yaml:54]:
```suggestion
        - --resource-type=gce-project,gpu-project,ingress-project,istio-project,scalability-presubmit-project,scalability-project,node-e2e-project,image-builder-aws-account
```

File: config/prow/cluster/boskos-reaper_deployment.yaml [hunk 2]
```diff
@@ -21,4 +21,4 @@ spec:
         image: gcr.io/k8s-staging-boskos/reaper:v20201002-ae3497d
         args:
         - --boskos-url=http://boskos.test-pods.svc.cluster.local.
-        - --resource-type=gce-project,gke-project,gpu-project,ingress-project,istio-project,scalability-presubmit-project,scalability-project,aws-account,node-e2e-project
+        - --resource-type=gce-project,gke-project,gpu-project,ingress-project,istio-project,scalability-presubmit-project,scalability-project,aws-account,node-e2e-project,image-builder-aws-project
```
Comment (detiber) [inline config/prow/cluster/boskos-reaper_deployment.yaml:24]:
```suggestion
        - --resource-type=gce-project,gke-project,gpu-project,ingress-project,istio-project,scalability-presubmit-project,scalability-project,aws-account,node-e2e-project,image-builder-aws-account
```

File: config/prow/cluster/boskos-resources.yaml [hunk 3]
```diff
@@ -575,3 +575,16 @@ resources:
...
+    - image-builder-aws-3
+    - image-builder-aws-4
+    - image-builder-aws-5
+    - image-builder-aws-6
+    - image-builder-aws-7
+    - image-builder-aws-8
+    - image-builder-aws-9
+    - image-builder-aws-10
+  state: dirty
+  type: image-builder-aws-project
```
Comment (detiber) [inline config/prow/cluster/boskos-resources.yaml:590]:
```suggestion
  type: image-builder-aws-account
```

File: config/prow/cluster/monitoring/mixins/lib/config.libsonnet [hunk 4]
```diff
@@ -68,7 +68,9 @@
       {instance: "104.197.27.114:9090", type: "node-e2e-project", friendly: "Node e2e project"},
       {instance: "104.197.27.114:9090", type: "scalability-project", friendly: "Scalability project"},
       {instance: "35.225.208.117:9090", type: "scalability-project", friendly: "Scalability project (k8s-infra)"},
-      {instance: "104.197.27.114:9090", type: "scalability-presubmit-project", friendly: "Scalability presubmit project"}
+      {instance: "104.197.27.114:9090", type: "scalability-presubmit-project", friendly: "Scalability presubmit project"},
+      {instance: "TODO:9090", type: "image-builder-aws-project", friendly: "Image Builder - AWS"},
```
Comment (detiber) [inline config/prow/cluster/monitoring/mixins/lib/config.libsonnet:72]:
```suggestion
      {instance: "TODO:9090", type: "image-builder-aws-account", friendly: "Image Builder - AWS"},
```

**Review by detiber — COMMENTED** [review]
Minor nit around the naming (account vs project) since boskos would be providing account credentials rather than a project, which is a GCP concept.

**Comment by kkeshavamurthy** [comment]
Thanks @detiber. Done.

File: config/prow/cluster/monitoring/mixins/lib/config.libsonnet [hunk 5]
```diff
@@ -68,7 +68,9 @@
       {instance: "104.197.27.114:9090", type: "node-e2e-project", friendly: "Node e2e project"},
       {instance: "104.197.27.114:9090", type: "scalability-project", friendly: "Scalability project"},
       {instance: "35.225.208.117:9090", type: "scalability-project", friendly: "Scalability project (k8s-infra)"},
-      {instance: "104.197.27.114:9090", type: "scalability-presubmit-project", friendly: "Scalability presubmit project"}
+      {instance: "104.197.27.114:9090", type: "scalability-presubmit-project", friendly: "Scalability presubmit project"},
+      {instance: "TODO:9090", type: "image-builder-aws-account", friendly: "Image Builder - AWS"},
```
Comment (detiber) [inline config/prow/cluster/monitoring/mixins/lib/config.libsonnet:72]: Maybe @dims could provide guidance on which boskos url we should target for this

Comment (dims) [reply config/prow/cluster/monitoring/mixins/lib/config.libsonnet:72]: better for @spiffxp ( or @bartsmykla ) to help with this. I am not quite sure how this was done before.

Comment (bartsmykla) [reply config/prow/cluster/monitoring/mixins/lib/config.libsonnet:72]: Unfortunately I don't have the knowledge about it either :-(

Comment (detiber) [reply config/prow/cluster/monitoring/mixins/lib/config.libsonnet:72]: If I had to guess one is related to the old google managed infrastructure and the other is related to the new wg-k8s-infra managed infrastructure, but I'm not sure as to where things are in the migration process

Comment (ixdy) [reply config/prow/cluster/monitoring/mixins/lib/config.libsonnet:72]: Hopefully this helps:

https://github.com/kubernetes/test-infra/blob/0d30bea7effaff7934b0ebce2e1c0b6335b125ba/config/prow/cluster/monitoring/additional-scrape-configs_secret.yaml#L13-L15

Comment (ixdy) [reply config/prow/cluster/monitoring/mixins/lib/config.libsonnet:72]: but given that you're adding these to the same boskos config that contains resource types such as `aws-account` and `node-e2e-project`, I think you want `104.197.27.114` here.

Comment (kkeshavamurthy) [reply config/prow/cluster/monitoring/mixins/lib/config.libsonnet:72]: Thanks for digging it up @ixdy

**Comment by detiber** [comment]
/hold 
blocked on https://github.com/kubernetes-sigs/boskos/pull/43
Also blocked on the secrets being created/loaded

**Comment by kkeshavamurthy** [comment]
@justinsb gentle reminder about sub account creation. Please let me know if I can help.

**Comment by codenrhoden** [comment]
@detiber Looks like the hold here can be lifted? With than and a new LGTM should be good to go.

**Comment by detiber** [comment]
@codenrhoden Yes, the boskos changes have already been made. I'm not sure about the actual account creation and the creation of the secrets, though. It is possible that those could still be blockers.

**Comment by spiffxp** [comment]
Have had priorities elsewhere, and still do for the next few days. Ping wg-k8s-infra on slack or add to next meeting agenda if still blocked by that time

**Comment by kkeshavamurthy** [comment]
@dims can you please help with this?

CC: @codenrhoden

**Comment by dims** [comment]
@kkeshavamurthy please take this to the k8s infra wg meeting like @spiffxp mentioned above.

**Pull request closed without merging**